static/src/javascripts/projects/common/modules/video/youtube-tracking.js

```diff
@@ -0,0 +1,98 @@
+define([
+    'common/utils/mediator',
+    'lodash/collections/forEach',
+    'common/utils/config',
+    'common/modules/video/ga-helper'
+], function (
+    mediator,
+    forEach,
+    config,
+    gaHelper
+) {
+
+    function eventAction() {
+        return 'video content';
+    }
+
+    function buildEventId(event, videoId) {
+        return event + ':' + videoId;
+    }
+
+    function initYoutubeEvents(videoId) {
+
+        var ga = window.ga;
+        var gaTracker = config.googleAnalytics.trackers.editorial;
+
+        var events = {
+            metricMap: {
+                'play': 'metric1',
+                'skip': 'metric2',
+                '25': 'metric3',
+                '50': 'metric4',
+                '75': 'metric5',
+                'end': 'metric6'
+            },
+            baseEventObject: {
+                eventCategory: 'Media',
+                eventAction: eventAction(),
+                eventLabel: videoId,
```

Comment:
@akash1810 See above comment.  VideoID is generic depending on underlying implementation.  i.e. It could support any type of videoId (youtube, atom, future id's?)